pkg/share/manager/jsoncs3/jsoncs3.go

```diff
@@ -697,3 +699,74 @@ func (m *Manager) UpdateReceivedShare(ctx context.Context, receivedShare *collab
 
 	return rs, nil
 }
+
+func shareIsRoutable(share *collaboration.Share) bool {
+	return strings.Contains(share.Id.OpaqueId, shareid.IDDelimiter)
+}
+func updateShareID(share *collaboration.Share) {
+	share.Id.OpaqueId = shareid.Encode(share.ResourceId.StorageId, share.ResourceId.SpaceId, share.Id.OpaqueId)
+}
+
+// Load imports shares and received shares from channels (e.g. during migration)
+func (m *Manager) Load(ctx context.Context, shareChan <-chan *collaboration.Share, receivedShareChan <-chan share.ReceivedShareWithUser) error {
+	log := appctx.GetLogger(ctx)
+	if err := m.initialize(); err != nil {
+		return err
+	}
+
+	var mu sync.Mutex
```

Comment:
As both go-routines operate on different data structures, they would not need to share the same lock, do they? The locks could be created within each go routine, then they would not block each other.

Reply:
hm, do I even need a mutex? the go routine will pull an element from the channel one by one. The caches Add() functions do *not* have their own lock ... so a lock in Load won't protect against concurrent updates via Share(), Unshare() or UpdateShare(), ... ah, no! it does not need to.

AFAICT the locks in Load are unnecessary. The CLI will instantiate its own instance of the manager and sync with the running share manager service using the cs3 backend.

anyway, I can drop the mutex completely from LOAD, can't I?

Reply:
Yes I think so. `Add` methods are only called by one go routine. If further syncing is needed I would build that directly into the `Add` method. But not needed for now.